invertedai/logs/debug_logger.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import json
+import os
+
+from collections import defaultdict
+from typing import List, Optional, Dict, Tuple
+from datetime import datetime, timezone
+
+logger = logging.getLogger(__name__)
+
+class DebugLogger:
+    def __init__(
+        self,
+        debug_log_path: str
+    ):
+        self.debug_log_path = debug_log_path
+        self._create_directory()
+
+        self.data = defaultdict(list)
+
+        file_name = "iai_log_" + self._get_current_time_human_readable_UTC() + "_UTC.json"
+        self.log_path = self.debug_log_path + file_name
+
+    def _get_current_time_human_readable_UTC(self):
+        return datetime.now(timezone.utc).strftime("%Y-%m-%d_%H:%M:%S:%f")
+
+    def _create_directory(self):
+        if not os.path.isdir(self.debug_log_path):
+            logger.info(f"Debug log directory does not exist: Created new directory at {self.debug_log_path}")
+            os.makedirs(self.debug_log_path)
+
+    def append_request(
+        self,
+        model: str,
+        data_dict: dict
```

Comment:
Not really important but the correct typing here and below should be `Optional[Dict]=None`

Reply:
In what situations would data be `None`? I think that change would be fine but I'm asking in case I'm wrong and there is more that should be done to handle that `None` case.